apple-codesign/src/bundle_signing.rs

```diff
@@ -199,7 +201,41 @@ impl SignedMachOInfo {
 
                 Some(format!("{}", req[0]))
             } else {
-                None
+                // In case no explicit requirements has been set, we use current file cdhashes.
+                let mut requirement_expr = None;
+
+                for macho in mach.iter_macho() {
+                    let signature = macho
+                        .code_signature()?
+                        .ok_or(AppleCodesignError::BinaryNoCodeSignature)?;
+
+                    let cd = if let Some(cd) = signature.code_directory_for_digest(DigestType::Sha256)? {
+                        cd
+                    } else if let Some(cd) = signature.code_directory_for_digest(DigestType::Sha1)? {
+                        cd
+                    } else if let Some(cd) = signature.code_directory()? {
+                        cd
+                    } else {
+                        return Err(AppleCodesignError::BinaryNoCodeSignature);
+                    };
```

Comment:
This looks very similar to the code a few lines above. I wonder if we should pull this out into a method on our `MachOBinary` struct...

Reply:
I moved it to its own method.